src/pages/workspace/WorkspaceNewRoomPage.js

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import {View} from 'react-native';
+import _ from 'underscore';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import withFullPolicy, {fullPolicyDefaultProps, fullPolicyPropTypes} from './withFullPolicy';
+import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
+import compose from '../../libs/compose';
+import HeaderWithCloseButton from '../../components/HeaderWithCloseButton';
+import Navigation from '../../libs/Navigation/Navigation';
+import ScreenWrapper from '../../components/ScreenWrapper';
+import styles from '../../styles/styles';
+import TextInputWithLabel from '../../components/TextInputWithLabel';
+import ExpensiPicker from '../../components/ExpensiPicker';
+import ONYXKEYS from '../../ONYXKEYS';
+import CONST from '../../CONST';
+import Button from '../../components/Button';
+import FixedFooter from '../../components/FixedFooter';
+import {createPolicyRoom} from '../../libs/actions/Report';
+
+const propTypes = {
+    /** All reports shared with the user */
+    reports: PropTypes.shape({
+        reportName: PropTypes.string,
+        type: PropTypes.string,
+        policyID: PropTypes.string,
+    }).isRequired,
+
+    ...fullPolicyPropTypes,
+
+    ...withLocalizePropTypes,
+};
+const defaultProps = {
+    betas: [],
+    ...fullPolicyDefaultProps,
+};
+
+class WorkspaceNewRoomPage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            roomName: '',
+            policyID: '',
+            visibility: CONST.REPORT.VISIBILITY.RESTRICTED,
+            isLoading: false,
+            error: '',
+            workspaceOptions: [],
+        };
+        this.onWorkspaceSelect = this.onWorkspaceSelect.bind(this);
+        this.onSubmit = this.onSubmit.bind(this);
+        this.checkAndModifyRoomName = this.checkAndModifyRoomName.bind(this);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.policies.length === prevProps.policies.length) {
+            return;
+        }
+        const workspaces = _.filter(this.props.policies, policy => policy && policy.type === CONST.POLICY.TYPE.FREE);
```

Comment:
maybe add a comment to the filter, I think I understand it, but it would be nice to know that I'm correct in my assumption, also to whoemver is readin this later.